### PR TITLE
Fixes #197:App crashes in MedicineStore Activity

### LINKED ISSUE #197: App crashes in MedicineStore Activity
**Current Behaviour**
If the user enters any big number in **Add medicine to store** dialog then app crashes .

ezgif com-optimize 3

**Logcat for the issue**
01-12 19:59:58.141 4508-4508/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                 Process: com.peacecorps.malaria, PID: 4508
                                                 java.lang.NumberFormatException: Invalid int: "744644464643377976767"
                                                     at java.lang.Integer.invalidInt(Integer.java:138)
                                                     at java.lang.Integer.parse(Integer.java:413)                                                 
                                                     at java.lang.Integer.parseInt(Integer.java:334)
                                                     at com.peacecorps.malaria.activities.MedicineStore$2$1.onClick(MedicineStore.java:167)
                                                     at android.view.View.performClick(View.java:5201)                                         
                                                     at android.os.Handler.handleCallback(Handler.java:746)
                                                     at android.os.Handler.dispatchMessage(Handler.java:95)
                                                     at android.os.Looper.loop(Looper.java:148)
                                                     at android.app.ActivityThread.main(ActivityThread.java:5443)
                                                     at java.lang.reflect.Method.invoke(Native Method)
     
**Expected behaviour**
Either we can introduce a check on this value so that user will be shown an error message if he crosses that particular limit  **or**
we can restrict the user to type the value upto a certain limit.

@chhavip @yatna I would like to ask,what approach should I follow either first--> to introduce a check or second -->to type value upto certain limit.

After the approval I will send the PR.Thanks :)

### PR DESCRIPTION
**Pull request by HimanshuS01**
Fixes #197 along with point number 2 in issue #216 

ezgif com-optimize 6

@chhavip @yatna @anubhakushwaha Please review this PR.Thanks :)

### CODE REVIEW

**Comment by free4murad** [comment]
@HimanshuS01 
You might wanna change the base branch.
I made the same mistake in my last 2 PRs.

**Comment by HimanshuS01** [comment]
@free4murad Thanks for telling. I have changed the base branch from master to develop. 
@yatna @chhavip @anubhakushwaha please review the PR.Thanks

**Comment by HimanshuS01** [comment]
@yatna I got your point and will do the suggested changes soon.Thanks

**Comment by HimanshuS01** [comment]
@yatna changes done.Please review it .Thanks :)

**Comment by HimanshuS01** [comment]
@yatna @anubhakushwaha @chhavip changes done.Please review .Thanks :)

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java [hunk 1]
```diff
@@ -215,13 +247,39 @@ public void onClick(View view) {
...
+                                    Integer.parseInt(time.getText().toString().trim()) > 100) {
+                                time.setError("Limit Exceeded");
+                                isLimitExceede = true;
+                            }
+                            else {
+                                isLimitExceede = false;
+                            }
+                        }
+                        catch (NumberFormatException exception) {
+                            time.setError("Limit Exceeded");
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: Put errors in strings.xml as it is used multiple times. Do for all errors denoted.

**Comment by HimanshuS01** [comment]
@chhavip I have made the requested changes. Please review it. Thanks.

**Comment by chhavip** [comment]
the base branch should be develop, please change the same  @HimanshuS01

**Comment by HimanshuS01** [comment]
@chhavip actually  when I changed my base branch to develop,All the previous commits(around 97 commits) in develop branch are showing in my PR along with my commit.
There is some problem arising when I update my origin/develop branch with the upstream/develop.
For reference I have used the following commands:
git fetch upstream
git merge upstream/develop
git checkout -b RangeIssue
then after commiting  my changes ,all the previous commits are showing in my PR.

**Comment by free4murad** [comment]
@HimanshuS01 
instead of `git merge upstream/develop` you need to use `git rebase upstream/develop`.
also force push your develop branch using `git push --force origin develop`

**Comment by HimanshuS01** [comment]
@chhavip I have changed the base branch to develop from master. Please review and merge this PR. Thanks :)

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java [hunk 2]
```diff
@@ -89,6 +92,29 @@ public void onClick(View view) {
...
                 //implement the email button
+
+                medicineQuantityEt.setOnKeyListener(new View.OnKeyListener() {
+
+                    @Override
+                    public boolean onKey(View v, int keyCode, KeyEvent event) {
+                        try {
+                            if (!medicineQuantityEt.getText().toString().equals("") &&
+                                    Integer.parseInt(medicineQuantityEt.getText().toString().trim()) > 100) {
+                                medicineQuantityEt.setError(String.valueOf(R.string.limit_exceeded));
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: @HimanshuS01 this method of referencing the string value is showing the value stored in R file,instead use getResources().getString(_id_)

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java [hunk 3]
```diff
@@ -89,6 +92,29 @@ public void onClick(View view) {
...
+                                isLimitExceede = true;
+                            }
+                            else {
+                                isLimitExceede = false;
+                            }
+                        }
+                        catch (NumberFormatException exception){
+                            medicineQuantityEt.setError(String.valueOf(R.string.limit_exceeded));
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: Same

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java [hunk 4]
```diff
@@ -98,6 +124,9 @@ public void onClick(View view) {
                         else if(medicineQuantityEt.getText().toString().matches("[0]+")){
                             medicineQuantityEt.setError("Quantity Required");
                         }
+                        else if(isLimitExceede){
+                            medicineQuantityEt.setError(String.valueOf(R.string.limit_exceeded));
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: Same

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java [hunk 5]
```diff
@@ -124,6 +153,9 @@ public void onClick(View view) {
...
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: Same

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java [hunk 6]
```diff
@@ -215,13 +247,39 @@ public void onClick(View view) {
...
+
+                time.setOnKeyListener(new View.OnKeyListener() {
+
+                    @Override
+                    public boolean onKey(View v, int keyCode, KeyEvent event) {
+
+                        try {
+                            if (!time.getText().toString().equals("") &&
+                                    Integer.parseInt(time.getText().toString().trim()) > 100) {
+                                time.setError(String.valueOf(R.string.limit_exceeded));
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: Same

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java [hunk 7]
```diff
@@ -215,13 +247,39 @@ public void onClick(View view) {
...
+                                isLimitExceede = true;
+                            }
+                            else {
+                                isLimitExceede = false;
+                            }
+                        }
+                        catch (NumberFormatException exception) {
+                            time.setError(String.valueOf(R.string.limit_exceeded));
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: Again

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java [hunk 8]
```diff
@@ -215,13 +247,39 @@ public void onClick(View view) {
...
+                });
                 set.setOnClickListener(new View.OnClickListener() {
                     @Override
                     public void onClick(View view) {
                         if(time.getText().toString().trim().equals("")){
                             //If nothing is entered display error
                             time.setError("Entry Required");
                         }
+                        else if(isLimitExceede){
+                            time.setError(String.valueOf(R.string.limit_exceeded));
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: Same

File: malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java (hunk 2 above)
Comment (HimanshuS01) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/activities/MedicineStore.java]: sure @anubhakushwaha :)

**Comment by HimanshuS01** [comment]
@anubhakushwaha changes done! Please merge this PR now.
thank you :)

**Pull request merged**